### ISSUE TITLE
regarding the result of DIAYN

### ISSUE THREAD

**Issue by lrubato** [artberryx/LSD#2]
Interesting work!

Can i also get a result of DAIYN using this codebase?

**Comment by seohongpark** [comment]
Thank you for your interest in this work. You can use the following code snippet in `_get_te_option_log_probs` (in `lsd.py`) (and turn off spectral normalization) to run DIAYN:

```
            target_dists = self.traj_encoder(v['next_obs_flat'])

            if self.discrete:
                logits = target_dists.mean
                te_option_log_probs = -torch.nn.functional.cross_entropy(logits, v['options_flat'].argmax(dim=1), reduction='none')
            else:
                te_option_log_probs = target_dists.log_prob(v['options_flat'])
```
